extensions/http-curl/client/HTTPStream.cpp

```diff
@@ -53,16 +53,16 @@ void HttpStream::seek(uint64_t offset) {
   throw std::exception();
 }
 
-int HttpStream::writeData(std::vector<uint8_t> &buf, int buflen) {
+int HttpStream::writeData(const std::vector<uint8_t> &buf, int buflen) {
   if ((int)buf.capacity() < buflen) {
     return -1;
   }
-  return writeData(reinterpret_cast<uint8_t *>(&buf[0]), buflen);
```

Comment:
this fails on some compilers. I spent a long time and saw failures on older versions ( not sure if specifically this one ). Did this cause a specific bug? These are non const for a reason and that's to tell the caller that their data may change. You're making an assumption that it shouldn't be const.